graylog2-server/src/main/java/org/graylog2/system/stats/elasticsearch/ElasticsearchProbe.java

```diff
@@ -16,107 +16,48 @@
  */
 package org.graylog2.system.stats.elasticsearch;
 
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.node.ArrayNode;
-import com.fasterxml.jackson.databind.node.JsonNodeType;
-import com.google.common.collect.Lists;
 import com.google.inject.Singleton;
-import io.searchbox.client.JestClient;
-import io.searchbox.client.JestResult;
-import io.searchbox.cluster.Health;
-import io.searchbox.cluster.PendingClusterTasks;
-import io.searchbox.cluster.Stats;
 import org.graylog2.indexer.IndexSetRegistry;
-import org.graylog2.indexer.cluster.jest.JestUtils;
+import org.graylog2.indexer.cluster.ClusterAdapter;
+import org.graylog2.indexer.cluster.PendingTasksStats;
+import org.graylog2.indexer.indices.HealthStatus;
 
 import javax.inject.Inject;
 import java.util.Arrays;
 import java.util.List;
-import java.util.Locale;
 
 @Singleton
 public class ElasticsearchProbe {
-    private final JestClient jestClient;
     private final IndexSetRegistry indexSetRegistry;
+    private final ClusterAdapter clusterAdapter;
 
     @Inject
-    public ElasticsearchProbe(JestClient jestClient, IndexSetRegistry indexSetRegistry) {
-        this.jestClient = jestClient;
+    public ElasticsearchProbe(IndexSetRegistry indexSetRegistry, ClusterAdapter clusterAdapter) {
         this.indexSetRegistry = indexSetRegistry;
+        this.clusterAdapter = clusterAdapter;
     }
 
     public ElasticsearchStats elasticsearchStats() {
```

Comment:
I think it make sense to move this method into `ClusterAdapter` (or create a separate interface which `ClusterAdapterES6` implements?) and get rid of `ElasticsearchProbe` completely. Right now `ElasticsearchProbe#elasticsearchStats` calls 4 different methods from `ClusterAdapter`, 3 of which are only used here. Those could become private, if the method were moved, thus reducing coupling.

The moved method should get the indices as a parameter to avoid a dependency on `IndexSetRegistry`, i.e. something like this: `ClusterAdapter#statsForIndices(List<String> indices)`.
It would imply that `org.graylog2.system.stats.ClusterStatsService` would need the dependency on `IndexSetRegistry`, but that seems acceptable to me.